model/src/guild/member.rs

```diff
@@ -74,6 +80,58 @@ impl MemberIntermediary {
     }
 }
 
+/// The state of a member's guild timeout.
```

Comment:
The documentation should highlight why the state is wrapped in a `MemberTimeoutState`.